topics/statistics/tutorials/regression_machinelearning/tutorial.md

```diff
@@ -0,0 +1,448 @@
+
+
+> ### {% icon details %} 5-fold cross-validation
+>
+> It is a model validation technique which estimates the performance of a predictive model on an unseen data. A dataset is divided into `5` folds and these folds are categorised into training and validation sets. The idea of cross-validation is shown in figure [3](#figure-3). The complete dataset is divided into `5` equal parts. 80% of the dataset is used for training and the remaining 20% is used for validating the performance of training. This is done for `5` folds/iterations, each time the validation set (20% of the dataset) is different. In all five folds, the complete dataset is used for training and validation. The final validation performance is averaged over `5` folds.
+>
+> ![5fold_cv](../../images/age-prediction-with-ml/5fold_cv.png "5-fold cross validation. ")
+>The image shows how the 5-fold cross-validation works. The complete dataset is divided into 5 equal parts/folds. 4 parts (80%) of the data (training set shown in yellow) are used for training the model and the remaining one part is used for evaluating (validation set shown in blue) the trained model. This is repeated for 5 times till every part/fold is used as the validation set. The accuracies computed for different validation folds are averaged to give 5-fold cross-validation accuracy.
+>
+{: .details}
+
+> ### {% icon hands_on %} Hands-on: Hyperparameter search
+>
+> 1. **Hyperparameter search** {% icon tool %} with the following parameters:
+>    - *"Select a model selection search scheme"*: `GridSearchCV - Exhaustive search over specified parameter values for an estimator `
+>        - {% icon param-files %} *"Choose the dataset containing pipeline/estimator object"*: `zipped` file (output of **Pipeline builder** {% icon tool %})
+>        - In *"Search parameters Builder"*:
+>             - {% icon param-files %} *"Choose the dataset containing parameter names"*: `tabular` file (output of **Estimator attributes** {% icon tool %})
+>             - In *"Parameter settings for search"*:
+>                 - {% icon param-repeat %} *"1: Parameter settings for search"*
+>                    - *"Choose a parameter name (with current value)"*: `gradiantboostingregressor_n_estimators: 100`
+>                    - *"Search list"*: `[25, 50, 75, 100, 200]`
+>        - In *"Advanced Options for SearchCV"*:
+>            - *"Select the primary metric (scoring)"*: `Regression -- 'r2'`
+>
+>               A scoring metric can be set. In this tutorial, we use `Regression -- 'r2'`
+>
+>            - *"Select the cv splitter"*: `KFold`
+>
+>               There are different ways to split the dataset into training and validation sets. In our tutorial, we will use `KFold` which splits the dataset into `K` consecutive parts. It is used for cross-validation. It is set to `5` using another parameter `n_splits`.
```

Comment:
consecutive parts for cross-validation. ?

Reply:
Yes, in cross-validation we need these parts